docs/navigation-advanced.md

```diff
@@ -0,0 +1,334 @@
+        customLogoutFn: myLogoutFn
+      },
+```
+
+
+## Context switcher
+
+<img src="https://github.com/SAP/luigi/blob/master/docs/assets/context-switcher.png" width="628">
+
+The context switcher is a drop-down element in the top navigation. It allows you to switch between a curated list of navigation elements such as Environments. To do so, add the contextSwitcher property to the navigation object. Find all the parameters you can use to configure it [here](navigation-parameters-reference.md#context-switcher).
+
+ Example:
+
+```javascript
+contextSwitcher: {
+      defaultLabel: 'Select Environment ...',
+      testId: 'myTestId',
+      parentNodePath: '/environments',
+      lazyloadOptions: false,
+      fallbackLabelResolver: (id) => (id.toUpperCase()),
+      options: [{label,pathValue}, {label,pathValue}],
+      actions: [{label,link,position,clickHandler?}]
+    },
+```
+
+## Product switcher
+
+<img src="https://github.com/SAP/luigi/blob/master/docs/assets/product-switcher.png" width="704">
+
+The product switcher is window in top the navigation which allows you to switch between navigation elements displayed there. To add it to your application, include the **productSwitcher** property in your **navigation** object. You may also add any of the parameters listed [here](navigation-parameters-reference.md#product-switcher).
+
+Example:
+```javascript
+productSwitcher: {
+      label: 'My Products',
+      testId: 'myTestId',
+      icon: 'grid',
+      items: [
+        {
+          icon: '',
+          label: 'Luigi in Github',
+          testId: 'myTestId',
+          externalLink: {
+            url: 'https://luigi-project.io/',
+            sameWindow: false
+          }
+        },
+        {
+          icon: '',
+          label: 'Project 1',
+          testId: 'myTestId',
+          link: '/projects/pr1'
+        }
+      ]
+    },
+```
+
+## App switcher
+
+<img src="https://github.com/SAP/luigi/blob/master/docs/assets/app-switcher.png" width="407">
+
+The app switcher is a drop-down in top the navigation which allows you to switch between applications. To use it, add the **addSwitcher** property in your **navigation** object. You may also add any of the parameters listed [here](navigation-parameters-reference.md#product-switcher).
+
+Example:
+
+```javascript
+appSwitcher = {
+    showMainAppEntry: true,
+    items: [
+      {
+        title: 'Application One',
+        subTitle: 'the first app',
+        link: '/projects/pr1'
+      },
+      {
+        title: 'Application Two',
+        link: '/projects/pr2',
+        subTitle: 'the second app'
+      },
+    ]
+  };
+```
+
+## Additional options
+
+For more options and parameters which you can use to configure navigation in Luigi, read the [full parameter reference](navigation-parameter-reference.md). Some of the topics you can find there include:
+
+* Enabling and disabling the [loading indicator](navigation-parameters-reference.md#loadingindicatorenabled)
```

Comment:
All links in this list transfer only to the `navigation-parameters-reference.md` page, but not to the particular sections.